### PR TITLE
Allow streaming of modified linked contracts in addition to the root contract

### PR DESCRIPTION
**Pull request by Ereski**
Change-type: minor
Signed-off-by: Carol Schulze <carol@balena.io>

### CODE REVIEW

**Comment by LucianBuzzo** [comment]
@Ereski Could you explain how this works? It's very hard to parse by looking at the code

**Comment by Ereski** [comment]
@LucianBuzzo the idea is that instead of saying "try the root contract with this specific ID" it says "try at least one of the root or any linked contract with this ID"

**Comment by Ereski** [comment]
@LucianBuzzo found the bug btw. Hopefully the new version will work and wont cause the query planner to go awry

File: lib/backend/postgres/streams.ts [hunk 1]
```diff
@@ -299,11 +299,12 @@ export class Stream extends EventEmitter {
 				this.cardTypes = _.compact(deversionedTypes);
 			}
 		}
+
 		this.streamQuery = Context.prepareQuery(
 			backend.compileSchema(this.context, this.streamer.table, select, schema, {
 				...options,
-				limit: 1,
-				extraFilter: `${this.streamer.table}.id = $1`,
+				limit: backend.MAXIMUM_QUERY_LIMIT,
```
Comment (LucianBuzzo) [inline lib/backend/postgres/streams.ts:306]: Why do we need the larger limit on the query?

Comment (Ereski) [reply lib/backend/postgres/streams.ts:306]: @LucianBuzzo there may be multiple root contracts that match the same linked contract. Which reminds me that I didn't modify the code to properly handle the case of >1 root contracts

**Comment by LucianBuzzo** [comment]
@Ereski I fixed the last few failing tests and things are looking food 👍

**Comment by LucianBuzzo** [comment]
I continued the work on [this branch](https://github.com/product-os/autumndb/pull/1462), as I wasn't sure if it would succeed, but looks like everything is working
